src/darjeeling/evaluator.py

```diff
@@ -175,16 +184,30 @@ def _evaluate(self, candidate: Candidate) -> CandidateOutcome:
 
         # select a subset of tests to use for this evaluation
         tests, remainder = self._select_tests()
+
         if not self.__run_redundant_tests:
             tests, redundant = self._filter_redundant_tests(candidate, tests)
         else:
             redundant = set()
 
+        heldout = set()
+        if not self.__run_heldout_tests:
+            heldout = self.heldout_tests    
+        valid_heldout = True if len(heldout)>0 else False
+        
```

Comment:
Definition of valid_heldout can be simplified:
```python
valid_heldout = bool(heldout)
```

Also, can we also get rid of 3/4 of the new lines to improve readability?